### PR TITLE
Adds pulumi kubernetes provider

### PR DESCRIPTION
**Pull request by DavidSeptimus-Klotho**
Resolves https://github.com/klothoplatform/klotho/issues/417

This PR primarily adds support for supplying a pulumi kubernetes provider to HelmChart resources via the IAC plugin. Each cluster will add its own Kubeconfig resource to the graph, which a chart can pull in as an IaCValue. Immediately after rendering the Kubeconfig resource, the TemplatesCompiler will also render a KubernetesProvider variable for that Kubeconfig, which can be resolved from the chart's ClustersProvider IaCValue.

Additional bugs related to dependencies were fixed in order to ensure that the final index.ts output was valid typescript.

### CODE REVIEW

File: pkg/infra/iac2/templates_compiler.go [hunk 1]
```diff
@@ -184,13 +185,16 @@ func (tc TemplatesCompiler) renderResource(out io.Writer, resource core.Resource
...
+		case "dependsOn":
 			inputArgs[fieldName] = stringTemplateValue{value: tc.resolveDependencies(resource)}
 			continue
-		}
-		if fieldName == "protect" {
+		case "protect":
 			inputArgs[fieldName] = stringTemplateValue{value: "protect", raw: "protect"}
 			continue
+		case "awsProfile":
+			inputArgs[fieldName] = stringTemplateValue{value: "protect", raw: "awsProfile"}
```
[the remark this answers is not in the document]
Comment (DavidSeptimus-Klotho) [reply pkg/infra/iac2/templates_compiler.go:196]: good catch.

File: pkg/infra/iac2/plugin.go [hunk 2]
```diff
@@ -44,7 +44,10 @@ func (p Plugin) Translate(cloudGraph *core.ResourceGraph) ([]core.File, error) {
 
 	buf.Write([]byte("export = async () => {\n"))
 	buf.Write([]byte("const kloConfig: pulumi.Config = new pulumi.Config('klo')\n"))
-	buf.Write([]byte("const protect = kloConfig.getBoolean('protect') ?? false\n\n"))
+	buf.Write([]byte("const protect = kloConfig.getBoolean('protect') ?? false"))
+	buf.Write([]byte(`
+const awsConfig = new pulumi.Config('aws')
+const awsProfile = awsConfig.get('profile')` + "\n\n"))
```
[the remark this answers is not in the document]
Comment (DavidSeptimus-Klotho) [reply pkg/infra/iac2/plugin.go:50]: no, it just gets it if it's set.

File: pkg/infra/kubernetes/kubeconfig.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	}
+)
+
+func (k Kubeconfig) Provider() string { return "kubernetes" }
+
+func (k Kubeconfig) KlothoConstructRef() []core.AnnotationKey { return k.ConstructsRef }
+
+func (k Kubeconfig) Id() string {
+	return fmt.Sprintf("%s:%s:%s", k.Provider(), KUBE_CONFIG_TYPE, k.Name)
+}
```
[the remark this answers is not in the document]
Comment (DavidSeptimus-Klotho) [reply pkg/infra/kubernetes/kubeconfig.go:54]: I've removed it from the graph

**Review by jhsinger-klotho — APPROVED** [review]

**Comment by jhsinger-klotho** [comment]
template failure for kubeconfig before you push

**Pull request merged**